src/Microsoft.DotNet.ProjectModel.Server/Program.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.ProjectModel.Resolution;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.ProjectModel.Server
+{
+    public class Program
+    {
+        private readonly Dictionary<int, ApplicationContext> _keepers = new Dictionary<int, ApplicationContext>();
+        private readonly WorkspaceContext _workspaceContext;
+        private readonly FrameworkReferenceResolver _frameworkReferenceResolver;
+        private readonly ProtocolManager _protocolManager;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly string _hostName;
+        private readonly int _port;
+        private Socket _listenSocket;
+
+        public Program(int intPort, string hostName, ILoggerFactory loggerFactory)
+        {
+            _port = intPort;
+            _hostName = hostName;
+            _loggerFactory = loggerFactory;
+            _protocolManager = new ProtocolManager(maxVersion: 4, loggerFactory: _loggerFactory);
+            _workspaceContext = WorkspaceContext.Create();
+            _frameworkReferenceResolver = new FrameworkReferenceResolver(ProjectContextBuilder.GetDefaultReferenceAssembliesPath());
```

Comment:
Maybe just add a singleton instance of this. FrameworkReferenceResolver.Default that does this